config/avahi/avahi.inc

```diff
@@ -203,9 +225,49 @@ function avahi_sync() {
 	}
 	avahi_write_config();
 	// Is package enabled?
-	if (($config['installedpackages']['avahi']['config'][0]['enable']) && file_exists("/usr/local/etc/rc.d/avahi-daemon.sh")) {
+	if ($config['installedpackages']['avahi']['config'][0]['enable']) {
 		start_service("avahi");
 	}
 }
 
+function avahi_upgrade_config() {
+	global $config, $avahi_config;
+
+	$avahi_config = $config['installedpackages']['avahi']['config'][0];
+	if (!is_array($avahi_config)) {
+		$avahi_config = array();
+	}
+	$changes = 0;
+	// Convert previous disable values to enable ones
+	if (isset($avahi_config['disable_ipv4']) && $avahi_config['disable_ipv4'] != "on") {
+		$avahi_config['enable_ipv4'] == "on";
+		$changes++;
+	}
+	if (isset($avahi_config['disable_ipv6']) && $avahi_config['disable_ipv6'] != "on") {
+		$avahi_config['enable_ipv6'] == "on";
+		$changes++;
+	}
+	// If broken settings were configured due to lack of validation, just set to defaults
+	if (isset($avahi_config['disable_ipv4']) && isset($avahi_config['disable_ipv6']) && $avahi_config['disable_ipv4'] == "on" && $avahi_config['disable_ipv6'] == "on") {
```

Comment:
It's still missing unset() calls for old config knobs, otherwise it's going to keep it in config unnecessarily